File: C++/Medium/WrongAnswer/optimal-partition-of-string_1.cpp
class Solution {
public:
    int partitionString(string s) {
        
        int n = s.size();
        vector<int> last(26, -1);
        for (int i = 0; i < n; ++i) {
            last[s[i] - 'a'] = i;
        }
        int res = 0;
        int l = 0, r = 0;
        for (int i = 0; i < n; ++i) {
            r = max(r, last[s[i] - 'a']);
            if (i == r) {
                ++res;
                l = i + 1;
            }
        }
        return res;
    }
};

File: Rust/Medium/WrongAnswer/optimal-partition-of-string_1.rs
impl Solution {
    pub fn partition_string(s: String) -> i32 {
        
        let mut ans = 0;
        let mut last = vec![0; 26];
        for (i, c) in s.bytes().enumerate() {
            last[(c - b'a') as usize] = i;
        }
        let mut start = 0;
        let mut end = 0;
        for (i, c) in s.bytes().enumerate() {
            end = end.max(last[(c - b'a') as usize]);
            if i == end {
                ans += 1;
                start = i + 1;
            }
        }
        ans
    }
}